### PR TITLE
Upgrade to geth 1.9.24

### PR DESCRIPTION
**Pull request by connorwstein**
https://github.com/ethereum/go-ethereum/releases/tag/v1.9.24

### CODE REVIEW

**Review by samsondav — DISMISSED** [review]
I NEED THIS

👍

File: core/internal/cltest/client.go [hunk 1]
```diff
@@ -160,8 +160,18 @@ func (c *SimulatedBackendClient) GetERC20Balance(address common.Address, contrac
 		return nil, errors.Wrapf(err, "while calling ERC20 balanceOf method on %s "+
 			"for balance of %s", contractAddress, address)
 	}
-	balance = new(big.Int)
-	return balance, balanceOfABI.Unpack(balance, "balanceOf", b)
+	out, err := balanceOfABI.Unpack("balanceOf", b)
```
Comment (connorwstein) [inline core/internal/cltest/client.go:163]: v1.9.24 API change

File: core/services/eth/contract.go [hunk 2]
```diff
@@ -30,7 +31,13 @@ func NewConnectedContract(
 	return &connectedContract{codec, address, ethClient, logBroadcaster}
 }
 
-func (contract *connectedContract) Call(result interface{}, methodName string, args ...interface{}) error {
+func (contract *connectedContract) Call(result interface{}, methodName string, args ...interface{}) (err error) {
```
Comment (connorwstein) [inline core/services/eth/contract.go:34]: A delicate change which affects FM jobs

File: core/services/eth/contract.go [hunk 3]
```diff
@@ -43,8 +50,15 @@ func (contract *connectedContract) Call(result interface{}, methodName string, a
...
-	err = contract.ABI().Unpack(result, methodName, rawResult)
-	return errors.Wrap(err, "unable to unpack values")
+	out, err := contract.ABI().Unpack(methodName, rawResult)
+	if err != nil {
+		return errors.Wrap(err, "unable to unpack values")
+	}
+	if len(out) != 1 {
+		return errors.Errorf("expected one unpacked value, got %d", len(out))
+	}
+	result = abi.ConvertType(out, result)
```
Comment (connorwstein) [inline core/services/eth/contract.go:60]: this thing warrants the panic catching

File: go.mod [hunk 4]
```diff
@@ -27,38 +22,29 @@ require (
...
 	github.com/pelletier/go-toml v1.8.1
 	github.com/pkg/errors v0.9.1
 	github.com/prometheus/client_golang v1.8.0
-	github.com/prometheus/tsdb v0.10.0 // indirect
 	github.com/robfig/cron/v3 v3.0.1
-	github.com/rs/cors v1.7.0 // indirect
 	github.com/satori/go.uuid v1.2.0
 	github.com/shopspring/decimal v1.2.0
-	github.com/smartcontractkit/libocr v0.0.0-20201203233047-5d9b24f0cbb5
+	github.com/smartcontractkit/libocr v0.0.0-20201209002813-4110928c10ff
```
Comment (connorwstein) [inline go.mod:42]: libocr now also is using v1.9.24

**Review by se3000 — APPROVED** [review]
💯

**Pull request merged**